### PR TITLE
Grafana: Replace magic number with a constant variable in response status

### LINKED ISSUE #80133: Grafana: Replace magic number with const var to response status
**What is the chore?**:

The http package's const variable and the magic number card are being mixed when the code indicates the status.

For the prevention of misrepresentation and uniformity, it would be better to use it only as a const variable. Therefore, I have worked on PR for this.

**Is there anything else we need to know?**:

Nope.

### PR DESCRIPTION
**Pull request by rlaisqls**
The http package's const variable and the magic number card are being mixed when the code indicates the status.

For the prevention of misrepresentation and uniformity, it would be better to use it only as a const variable. Therefore, I have worked on PR for this.

Which issue(s) does this PR fix?:

Fixes #80133 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/dashboardimport/api/api.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func (api *ImportDashboardAPI) ImportDashboard(c *contextmodel.ReqContext) respo
 	}
 
 	if limitReached {
-		return response.Error(403, "Quota reached", nil)
+		return response.Error(http.StatusInternalServerError, "Quota reached", nil)
```
Comment (papagian) [inline pkg/services/dashboardimport/api/api.go:74]:
```suggestion
		return response.Error(http.StatusForbidden, "Quota reached", nil)
```

File: pkg/services/libraryelements/api.go [hunk 2]
```diff
@@ -309,31 +309,31 @@ func (l *LibraryElementService) filterLibraryPanelsByPermission(c *contextmodel.
...
-		return response.Error(412, model.ErrLibraryElementVersionMismatch.Error(), err)
+		return response.Error(http.StatusPreconditionFailed, model.ErrLibraryElementVersionMismatch.Error(), err)
 	}
 	if errors.Is(err, dashboards.ErrFolderNotFound) {
-		return response.Error(404, dashboards.ErrFolderNotFound.Error(), err)
+		return response.Error(http.StatusNotFound, dashboards.ErrFolderNotFound.Error(), err)
 	}
 	if errors.Is(err, dashboards.ErrFolderAccessDenied) {
-		return response.Error(403, dashboards.ErrFolderAccessDenied.Error(), err)
+		return response.Error(http.StatusUnauthorized, dashboards.ErrFolderAccessDenied.Error(), err)
```
Comment (papagian) [inline pkg/services/libraryelements/api.go:327]:
```suggestion
		return response.Error(http.StatusForbidden, dashboards.ErrFolderAccessDenied.Error(), err)
```

**Review by papagian — CHANGES_REQUESTED** [review]
thank you for your contribution! I have spotted some inconsistencies

**Comment by rlaisqls** [comment]
I'm sorry I missed something... :( I fixed it now

**Review by papagian — APPROVED** [review]

**Review by IevaVasiljeva — APPROVED** [review]
I've skimmed over the changes, and it looks good to me! Thanks for the refactor!

Are we happy to merge this now @papagian?

**Comment by papagian** [comment]
> Are we happy to merge this now @papagian?

thanks for checking! there are some missing imports that cause the CI to fail: we need to fix them before merging

**Comment by IevaVasiljeva** [comment]
Yup, you're right @papagian, was just about to message about it - looks like you need to import the `http` package in `routing.go` @rlaisqls 
```pkg/api/routing/routing.go:11:25: undefined: http```

**Pull request merged**